app.js

```diff
@@ -0,0 +1,14 @@
+function parseFile(filename){
+    d3.csv(filename, function (data) {
+        return data;
+    }).then( function (data) {
+    // do something with data
+        console.log(data);
+    });
+}
+
+let submitFileNameBtn = document.getElementById('submitFileName');
+submitFileNameBtn.addEventListener('click', function getFileInput() {
+    let filename = document.getElementById("fileInput").value;
+    parseFile(filename);
+}, false);
```

Comment:
You should utilize some of the D3 helpers/syntax for this part! Something like:
`d3.select("#submitFileName").on("click", function(event){ <THE FUNCTION> })`
You could probably also use some D3 for getting the text from fileInput too, but it isn't immediately coming to me.
I referenced the d3-selection github documentation: https://github.com/d3/d3-selection

Reply:
- [ ] Use more D3 functions for this block.

Reply:
I replaced the event listener with the native d3 click detector, but getting the value of the html, at least from what I've read on google, is just plain complicated.